Code/Framework/AzCore/AzCore/IO/Streamer/StreamerContext.h

```diff
@@ -7,126 +7,127 @@
  */
 #pragma once
 
-#include <AzCore/base.h>
-#include <AzCore/IO/Streamer/FileRequest.h>
 #include <AzCore/IO/Streamer/Statistics.h>
 #include <AzCore/IO/Streamer/StreamerConfiguration.h>
 #include <AzCore/IO/Streamer/StreamerContext_Platform.h>
+#include <AzCore/Statistics/RunningStatistic.h>
+#include <AzCore/base.h>
 #include <AzCore/std/containers/deque.h>
-#include <AzCore/std/containers/vector.h>
 #include <AzCore/std/containers/queue.h>
-#include <AzCore/Statistics/RunningStatistic.h>
+#include <AzCore/std/containers/vector.h>
 
-namespace AZ
+namespace AZ::IO
 {
-    namespace IO
+    class FileRequest;
+    class ExternalFileRequest;
+
+    using FileRequestPtr = AZStd::intrusive_ptr<ExternalFileRequest>;
```

Comment:
Well this will cause a problem if both `FileRequests.h` and `StreamerContext.h` was included in the same file at any point.

If anything you could use an `inline namespace` in both files to distinguish the file aliases.
That way if there is ever a situation where both files need to be included into the same location, the developer can resolve the `FileRequestPtr` by explicitly indicating the namespace

Ex. File with either StreamerContext.h or FileRequest.h `include`ed
```
void Foo(AZ::IO::FileRequestPtr);
```

Ex. File with both StreamerContext.h and FileRequest.h `include`ed

```
// Pick one of the type aliases to resolve ambiguity
void Foo(AZ::IO::StreamerContextInternal::FileRequestPtr);
```

Reply:
I don't believe it will cause problems, since multiple `using` with the same type/name are allowed?
<https://godbolt.org/z/Ee8eofzYx>

Reply:
You are [correct](https://en.cppreference.com/w/cpp/language/typedef) about a type alias being able to re-declared to the same type.
> The typedef-names are aliases for existing types, and are not declarations of new types. Typedef cannot be used to change the meaning of an existing type name (including a typedef-name). Once declared, a typedef-name may only be redeclared to refer to the same type again